### DISCUSSION TITLE
Storybook behind Proxy: Reload Loop in GitHub Codespaces

### DISCUSSION THREAD

**Discussion by LarsHadidi** [storybookjs/storybook#26038] [Help]
### Summary

Storybook on SvelteKit in GitHub CodeSpace:
- Reload Loop due to failing WSS connection to a GitHub Codespace
Looks like #20068  that is supposed to be resolved

### Additional information

- `npm create svelte@latest` &rarr; create-svelte version 6.0.8
- `npx storybook init`&rarr; storybook@7.6.15
- `npm run storybook` &rarr; running storybook dev -p 6006 --quiet

### INFINITE LOOP
![image](https://github.com/storybookjs/storybook/assets/12017203/aa05aeae-c937-417a-830d-dd07a7bbd31b)

**Comment by MrSquaare** [comment]
Still a problem with Storybook 8.1.2

**Comment by jarda-svoboda** [comment]
Try this setting (if you are using Vite builder) in your `.storybook/main.js` (or `.ts`).

```jsonc
{
  //...

  async viteFinal(config) {
    // Merge custom configuration into the default config
    const { mergeConfig } = await import('vite');

    return mergeConfig(config, {
      server: {
        hmr: {
          clientPort: process.env.CODESPACES ? 443 : undefined, // <== this is the magic
        }
      }
    });
  },
}
```

Related issues: https://github.com/vitejs/vite/discussions/5195

**Reply by oldbig** [reply]
Hi @jarda-svoboda 
You saved me, thank you!
